doc/specs/#147 - Support different release channels [released, beta, alpha].md

```diff
@@ -0,0 +1,259 @@
+
+If the package channel cannot be determined, there are several options. 
+1) Ask the user for input.
+2) Throw a warning and skip that package
+3) Use the default channel for that package
+
+The Windows Package Manager should never leave a package in a state where it cannot be upgraded simply because the package is unknown. To account for this, there must be some way for the user to specify a release channel. Therefore, when `winget upgrade --all` is used, and the release channel cannot be determined, a warning should be shown and the package should be skipped. For all other cases, the user should be asked to select a release channel on a per-package basis; the user's choice for that package should persist.
+
+#### winget export/import
+When exporting the list of installed packages, where possible, the release channel should be included with the export.
+When importing the list of installed packages, where possible, the release channel should be honored. If the release channel is not able to be honored, a warning should be thrown, and the default release channel should be used
+
+### Settings
+The user should be able to specify a set of preferred release channels in their settings file. The `channel` parameter should be added under `installBehavior`. The matching parameter is `--channel`. The order of the preferences should be considered when installing packages such that the first value is the highest preference.
+```json
+"installBehavior": {
+        "preferences": {
+            "locale": [ "en-US", "fr-FR" ],
+            "channel": ["alpha","beta","dev","canary","nightly"]
+        }
+    }
+```
+
+## UI/UX Design
+### Searching for package version with release channel
+
+```raw
+PS> winget search Google.Chrome
+
+Name                       Id                         Version        Channel
+-----------------------------------------------------------------------------
+Google Chrome              Google.Chrome              96.0.4664.45   Stable
+Chrome Remote Desktop Host Google.ChromeRemoteDesktop 96.0.4664.39   Stable
+
+PS> winget search Google.Chrome --channel dev
+
+Name           Id             Version        Channel
+----------------------------------------------------
+Google Chrome  Google.Chrome  97.0.4692.20   Dev
+
+PS> winget search Google --channel dev
+Name           Id                   Version        Channel
+-----------------------------------------------------------
+Google Chrome  Google.Chrome        97.0.4692.20   Dev
+Android Studio Google.AndroidStudio 2020.3.1       dev
+```
+
+### Showing available release channels
+```raw
+# When all channels have defined names
+PS> winget show Google.Chrome --channels
+Found Google Chrome [Google.Chrome]
+Channel
+-------
+Stable
+Beta
+Canary
+Dev
+
+# When the default channel is unnamed
+PS> winget show 7zip.7zip --channels
+Found 7-zip [7zip.7zip]
+Channel
+-------
+Default
```

Comment:
If the default channel is always "empty", then there is no need to have a reserved value or even show it in this case.